src/main/java/thrift/logic/parser/ListCommandParser.java

```diff
@@ -18,25 +19,44 @@
+
+        if (!arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)
+                && !args.isEmpty()) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    ListCommand.MESSAGE_USAGE)); //prefix m/ is not present so there are invalid arguments
+        }
+
         return getCommand(argMultimap);
     }
 
     private ListCommand getCommand(ArgumentMultimap argMultimap) throws ParseException {
-        Optional<String> month = argMultimap.getValue(CliSyntax.PREFIX_MONTH);
-        Optional<String> tag = argMultimap.getValue(CliSyntax.PREFIX_TAG);
-        if (month.isEmpty() && tag.isEmpty()) {
-            return new ListCommand(); //list all transactions
-        } else if (!month.isEmpty() && !tag.isEmpty()) {
-            return new ListCommand(); //filter by both month and tag, coming in v1.3
-        } else if (!month.isEmpty() && tag.isEmpty()) {
-            return new ListCommand(); //filter by month, coming in v1.3
-        } else if (month.isEmpty() && !tag.isEmpty()) {
-            return new ListCommand(); //filter by tag, coming in v1.3
-        } else { //bad args input or wrong prefixes used will throw the ParseException
-            throw new ParseException(
-                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE));
+        if (arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)
+                && !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    ListCommand.MESSAGE_USAGE)); //prefix m/ exists, but there is invalid preamble before the prefix.
+        } else if (arePrefixesPresent(argMultimap, CliSyntax.PREFIX_MONTH)) {
+            try {
+                Month month = ParserUtil.parseMonth(argMultimap.getValue(CliSyntax.PREFIX_MONTH).get()); //to be used
+            } catch (Exception e) {
+                throw new ParseException(String.format(Messages.MESSAGE_INVALID_MONTH_FORMAT,
+                        ListCommand.MESSAGE_MONTH));
+            }
+            return new ListCommand(); //filter by month tbc, to look like return new ListCommand(month);
+        } else {
+            return new ListCommand(); //list all transactions, as prefix m/ does not exist.
         }
     }
 
+    /**
+     * This methods checks if the required prefixes are present in the {@code ArgumentMultimap}.
+     *
+     * @param argumentMultimap the object to check for the existence of prefixes.
+     * @param prefixes variable amount of {@code Prefix} to confirm the existence of.
+     * @return true if specified prefixes are present in the argumentMultimap.
+     */
+    protected static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
Do you really need this method to check for `/m` only though. Consider using `argumentMultimap.getSingleValue(CliSyntax.PREFIX_MONTH).isPresent()` to check for existence of `CliSyntax.PREFIX_MONTH` and that it only appears ONCE.

In your `ListCommandParser` class:
```
Optional<String> monthString = argumentMultimap.getSingleValue(CliSyntax.PREFIX_MONTH);
boolean isMonthPresent = monthString.isPresent();
```

This keeps your code cleaner without the repetitive long chain of calls to retrieve the same String from `argumentMultimap`.

Reply:
i think ill keep this as it is first? i think that i do need the areprefixes present when i extend the command to take in t/tags as well uh..

Reply:
Alright no worries.